tests/tck/features/match/PipeAndVariable.feature

```diff
@@ -60,3 +60,18 @@ Feature: Pipe or use variable to store the match results
       | 42  | false | "Tim Duncan" | "Shaquile O'Neal"   |
       | 42  | false | "Tim Duncan" | "Tiago Splitter"    |
       | 42  | true  | "Tim Duncan" | "Tony Parker"       |
+
+  Scenario: yield collect
+    When executing query:
+      """
+      MATCH (v:player)
+      WHERE v.name CONTAINS 'Tim'
+      RETURN v.age as age, id(v) as vid |
+      GO FROM $-.vid OVER like REVERSELY YIELD $-.age AS age, like._dst AS dst |
+      YIELD
+        any(d IN COLLECT(DISTINCT $-.dst) WHERE d=='Tony Parker') AS d,
+        $-.age as age
+      """
+    Then the result should be, in any order:
```

Comment:
IMO, `cypher` is like a superset of 'nGql', so it’s better to treat them as two different languages.

Reply:
Agree with you. When we support more powerful `MATCH`, we don't need to pipe the result of `MATCH` to nGQL. But this's a use case of nebula user, so I add it to CI for keeping it always right.

Reply:
@Shylock-Hg Maybe we need to support the input variables in MATCH since cypher has parameter syntax which is like nGQL variable in future. Even then, I still don't want to use them together :)

Reply:
I means that should add the failed case for input of match.

Reply:
I know. There are many scenarios to test for disabling input variable of MATCH, such as, in pattern or where clause,  so it's better to open a new PR.